tasks/utilities/data_handling/task_fastq_handeling.wdl

```diff
@@ -0,0 +1,47 @@
+version 1.0
+
+task sra_lite_autodetect {
+  meta {
+    description: "Detect if SRA-Lite FASTQ file"
+  }
+  input {
+    File read1
+    File? read2
+    String docker = "us-docker.pkg.dev/general-theiagen/quay/ubuntu:latest"
+    Int disk_size = 100
+    Int cpu = 1
+    Int memory = 2
+  }
+  command <<<
+    # determine if paired-end or not
+    if ! [ -z ~{read2} ]; then
+      echo "Reads are paired..."
+    fi
+
+    # detect if file is compressed and define the appropriate command
+    if [[ ~{read1} == *.gz ]]; then
+      echo "Reads are compressed..."
+      command="zcat"
+    else
+      echo "Reads are not compressed..."
+      command="cat"
+    fi
+
+    # check if the first quality control string is set to SRA-Lite
+    # SRA-Lite filetype has all the quality enconding set to the '?' character
+    # corresponding to a phred-score of 30
+    $command ~{read1} | head -n 4 | awk 'NR==4 {if ($0 ~ /^[?]+$/) {print "SRA-Lite FASTQ detected"} else {print ""}}' > WARNING
```

Comment:
I popped the `awk` command into chatgpt and if it's is interpreting the command correctly, then this `awk` statement looks for one or more question marks `?`s from the start-to-end of the 4th line, which would falsely flag FASTQ files that are formatted normally, but have at least one base with a Q30 score.

So I just want to double check it's behaving as expected on normal FASTQs too.

Reply:
`awk 'NR==4 {if ($0 ~ /^[?]+$/)`  :
- `NR==4` on the fourth line, 
- `if $0 ~ /.../` this will evaluate to true only if the entire line matches this regular expression, which is:
- `^` at the start of the line
- `[?]` look for this character
- `+$` and only this character until the end of the line

All together, this means that it's looking for a line consisting only of `?`s.

Reply:
...I think

Reply:
It should be checking if the 4th line has only `?` character! :)